### PR TITLE
Fix/sip 1066 non existing route id 500 error

### LINKED ISSUE #109: [BUG] Un-existing routeId causes 500 in /actuator/adapter-routes/{routeId}
**Describe the bug**
GET request for invalid routeId returns HTTP 500 with stack trace

**To Reproduce**
- An adapter is up and running, with defined routes
- Try to reach a non-existing route via Postman, e.g. "GET http://localhost:8081/actuator/adapter-routes/glgl " --> HTTP 500 is received and stack trace is shown

**Expected behavior**
- GET request for invalid routeId should return "404 Not found"

**SIP (please complete the following information):**
 - Involved components: Core

### PR DESCRIPTION
**Pull request by LetoBukarica**
# Description

This fix resolves an issue that occurs when the adapter-routes endpoint (actuator exposed endpoints) is called with an non-existing route. 500 error was thrown, now the error is handled better.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Context tests (MVC) are added. This fix was developed in a TDD manner.

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing (regression) unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java [hunk 1]
```diff
@@ -98,6 +101,11 @@ public void startAll() {
   @GetMapping("/{routeId}")
   @Operation(summary = "Get route details", description = "Get route details")
   public AdapterRouteDetails route(@RouteIdParameter @PathVariable String routeId) {
+    ManagedRouteMBean routeMBean = mbeanContext.getManagedRoute(routeId);
+    if (routeMBean == null) {
+      throw new ResponseStatusException(
+          HttpStatus.NOT_FOUND, "Route with ID:" + routeId + " not found");
+    }
     return new AdapterRouteDetails(mbeanContext.getManagedRoute(routeId));
```
Comment (vladiIkor) [inline sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java]:
```suggestion
    return AdapterRouteDetails(routeMBean);
```

File: sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java [hunk 2]
```diff
@@ -98,6 +101,11 @@ public void startAll() {
   @GetMapping("/{routeId}")
   @Operation(summary = "Get route details", description = "Get route details")
   public AdapterRouteDetails route(@RouteIdParameter @PathVariable String routeId) {
+    ManagedRouteMBean routeMBean = mbeanContext.getManagedRoute(routeId);
+    if (routeMBean == null) {
+      throw new ResponseStatusException(
+          HttpStatus.NOT_FOUND, "Route with ID:" + routeId + " not found");
```
Comment (vladiIkor) [inline sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java]: Let's use translations for messages. Especially it they are returned trough API

File: sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java [hunk 3]
```diff
@@ -133,7 +141,12 @@ public void resetAll() {
   @PostMapping("/{routeId}/reset")
   @Operation(summary = "Reset route", description = "Reset route")
   public void resetStatistics(@RouteIdParameter @PathVariable String routeId) {
-    mbeanContext.getManagedRoute(routeId).reset();
+    ManagedRouteMBean routeMBean = mbeanContext.getManagedRoute(routeId);
+    if (routeMBean == null) {
```
Comment (vladiIkor) [inline sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java]: can we reuse this validation block from above?

**Review by vladiIkor — DISMISSED** [review]

**Comment by vladiIkor** [comment]
Please check Checklist in PR form :)

File: sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java (hunk 3 above)
Comment (LetoBukarica) [reply sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java]: Take a look at the new version. I've wrapped that in a function and replaced all occurrences of mbeanContext.getManagedRoute through the new function. The question is if 404 makes sense for other functions. It is hardly possible to get a null on something like this one:
camelContext.getRoutes().forEach(route -> getRouteMBean(route.getRouteId()).reset());
But take a look and see if it makes sense.

**Review by Dzuci — DISMISSED** [review]

File: sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java (hunk 2 above)
Comment (LetoBukarica) [reply sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java]: Done

Comment (LetoBukarica) [reply sip-core/src/main/java/de/ikor/sip/foundation/core/actuator/routes/AdapterRouteEndpoint.java]: You are right, changed.

**Review by Dzuci — APPROVED** [review]

**Pull request merged**